be/src/Project/Web/Seller/Seller.Web/Areas/Products/ApiControllers/ProductsApiController.cs

```diff
@@ -43,12 +50,65 @@ public class ProductsApiController : BaseApiController
                 CultureInfo.CurrentUICulture.Name,
                 searchTerm,
                 hasPrimaryProduct,
-                GuidHelper.ParseNullable((this.User.Identity as ClaimsIdentity).Claims.FirstOrDefault(x => x.Type == AccountConstants.Claims.OrganisationIdClaim)?.Value),
+                GuidHelper.ParseNullable((User.Identity as ClaimsIdentity).Claims.FirstOrDefault(x => x.Type == AccountConstants.Claims.OrganisationIdClaim)?.Value),
                 pageIndex,
                 itemsPerPage,
                 null);
 
-            return this.StatusCode((int)HttpStatusCode.OK, products);
+            return StatusCode((int)HttpStatusCode.OK, products);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetProductOrderSuggestion(
+            string searchTerm,
+            bool? hasPrimaryProduct,
+            int pageIndex,
+            int itemsPerPage)
+        {
+            var token = await HttpContext.GetTokenAsync(ApiExtensionsConstants.TokenName);
+            var language = CultureInfo.CurrentUICulture.Name;
+
+            var products = await _productsRepository.GetProductsAsync(
+                token,
+                language,
+                searchTerm,
+                hasPrimaryProduct,
+                GuidHelper.ParseNullable((User.Identity as ClaimsIdentity).Claims.FirstOrDefault(x => x.Type == AccountConstants.Claims.OrganisationIdClaim)?.Value),
+                pageIndex,
+                itemsPerPage,
+                null);
+
+            var onStockProducts = await _inventoryRepository.GetInventoryProductsAsync(
```

Comment:
**suggestion:** it will be more optimal to get products from the warehouse that come from suggestions - use products.select(x => x.Id);